test/test_checks.py

```diff
@@ -21,24 +19,53 @@ def test_get_noqa_lines(self):
         self.assertEqual(checker.get_noqa_lines(checker.get_file_contents()), [2])
 
 
-class TestFlake8Stdin(TestCase):
-    @expectedFailureIf(FLAKE8_VERSION == '3')
+class TestFlake8(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        if version_info < (3, 0):
+            cls.expected_error_suffix = b'25: Q000 Remove bad quotes.'
+            # For some reason using "--select=Q" did suppress all outputs, so
+            # the result might contain non flake_quotes related errors
+            parameters = []
+        else:
+            cls.expected_error_suffix = b'24: Q000 Remove bad quotes.'
+            # Conversely Flake8 3.x currently has a bug that it doesn't report
+            # any error unless selected specifically
+            # See also: https://gitlab.com/pycqa/flake8/issues/183
+            parameters = ['--select=Q']
+        cls.command = ['flake8'] + parameters
+
+    def test_file(self):
+        """Test using stdin."""
+        filepath = get_absolute_path('data/doubles.py')
+        p = subprocess.Popen(self.command + [filepath],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Exactly: This is testing `flake8-quotes` in Flake8. If this test existed sooner we might have seen sooner that we broke support for Flake8 3.x. But we didn't noticed it, because the only Flake8 related test was expected to fail. So I thought that only Flake8 3.x using stdin was broken wen in fact Flake8 3.x was broken completely.

Now you could argue that the stdin test is no expected to succeed with your patch, but I'd argue against that because it is only testing stdin. The normal way could still be broken.

I don't specifically test that it can detect the one file correctly. It is just one randomly chosen file, but it tests that the plugin correctly works in Flake8.

And in theory it could be expanded to actually test more things that might break, like applying a setting (so we test that `--quotes`/`--inline-quotes` are actually applied) or selecting/ignoring the error code specifically (now that we made the plugin actually compatible) or using a configuration file (which might have shown us that using `type="choice"` was breaking something).

I'm fine with your #46 patch but I personally think that this test should be added (albeit improved to be not as “messy”).